app/controllers/products_controller.rb

```diff
@@ -2,27 +2,28 @@ class ProductsController < ApplicationController
   include ProductsHelper
 
   before_action :set_product, only: %i[ show edit update destroy ]
+  before_action :set_categories, only: [:new, :create, :edit, :update]
   rescue_from ActiveRecord::RecordNotFound, with: :invalid_product
   
-  # GET /products or /products.json
   def index
     @products = Product.all.order(:title)
+
+    respond_to do |format|
+      format.html
+      format.json
```

Comment:
what will be the output for this?

Reply:
This will send response as json rendered by file ```app/views/products/index.json.html```